src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -46832,6 +46832,1410 @@ class CL1
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "x").WithLocation(16, 9)
+                );
+        }
+
+        ///<remarks>Ported from <see cref="FlowTests.NullCoalescing_CondAccess_UserDefinedConv_02"/>.</remarks>
+        [Theory]
+        [InlineData("explicit")]
+        [InlineData("implicit")]
+        public void NullCoalescing_CondAccess_ExplicitUserDefinedConv_02_01(string conversionKind)
+        {
+            var source = @"
+class B { }
+class C
+{
+    public static " + conversionKind + @" operator B(C? c) => new B();
+
+    void M1(C c1, object? x)
+    {
+        B b = (B)c1?.M1(x = 0) ?? c1!.M2(x = 0);
+        x.ToString(); // 1
+    }
+
+    C M1(object obj) { return this; }
+    B M2(object obj) { return new B(); }
+}
+";
+            CreateNullableCompilation(source).VerifyDiagnostics(
+                // (10,9): warning CS8602: Dereference of a possibly null reference.
+                //         x.ToString(); // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "x").WithLocation(10, 9));
+        }
+
+        ///<remarks>Ported from <see cref="FlowTests.NullCoalescing_CondAccess_UserDefinedConv_02"/>.</remarks>
+        [Theory]
+        [InlineData("explicit")]
+        [InlineData("implicit")]
+        public void NullCoalescing_CondAccess_ExplicitUserDefinedConv_02_02(string conversionKind)
+        {
+            var source = @"
+class B { }
+class C
+{
+    public static " + conversionKind + @" operator B?(C? c) => new B();
+
+    void M1(C c1, object? x)
+    {
+        B b = (B?)c1?.M1(x = 0) ?? c1!.M2(x = 0);
+        x.ToString(); // 1
+    }
+
+    C M1(object obj) { return this; }
+    B M2(object obj) { return new B(); }
+}
+";
+            CreateNullableCompilation(source).VerifyDiagnostics(
+                // (10,9): warning CS8602: Dereference of a possibly null reference.
+                //         x.ToString(); // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "x").WithLocation(10, 9));
+        }
+
+        ///<remarks>Ported from <see cref="FlowTests.NullCoalescing_CondAccess_UserDefinedConv_02"/>.</remarks>
+        [Theory]
+        [InlineData("explicit")]
+        [InlineData("implicit")]
+        public void NullCoalescing_CondAccess_ExplicitUserDefinedConv_02_03(string conversionKind)
+        {
+            var source = @"
+using System.Diagnostics.CodeAnalysis;
+
+class B { }
+class C
+{
+    [return: NotNullIfNotNull(""c"")]
+    public static " + conversionKind + @" operator B?(C? c) => new B();
+
+    void M1(C c1, object? x)
+    {
+        B b = (B?)c1?.M1(x = 0) ?? c1!.M2(x = 0);
+        x.ToString(); // 1
+    }
+
+    C M1(object obj) { return this; }
+    B M2(object obj) { return new B(); }
+}
+";
+            CreateNullableCompilation(new[] { source, NotNullIfNotNullAttributeDefinition }).VerifyDiagnostics(
+                // (13,9): warning CS8602: Dereference of a possibly null reference.
```

Comment:
I think this a perfect example of the compromise of stricter methods here, and if we want to solve it for user-defined conversions, the equivalent static method should be solved as well :)

Reply:
Related to #36164